plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -0,0 +1,367 @@
+ *
+ * </pre>
+ * 
+ * @see azkaban.jobtype.HadoopSecureSparkWrapper
+ */
+public class HadoopSparkJob extends JavaProcessJob {
+
+  // Azkaban/Java params
+  private static final String HADOOP_SECURE_SPARK_WRAPPER = HadoopSecureSparkWrapper.class
+          .getName();
+
+  // Spark params
+
+  public static final String DRIVER_JAVA_OPTIONS = "driver-java-options";
+
+  // security variables
+  private String userToProxy = null;
+
+  private boolean shouldProxy = false;
+
+  private boolean obtainTokens = false;
+
+  private HadoopSecurityManager hadoopSecurityManager;
+
+  public HadoopSparkJob(String jobid, Props sysProps, Props jobProps, Logger log) {
+    super(jobid, sysProps, jobProps, log);
+
+    getJobProps().put(CommonJobProperties.JOB_ID, jobid);
+
+    shouldProxy = getSysProps().getBoolean(ENABLE_PROXYING, false);
+    getJobProps().put(ENABLE_PROXYING, Boolean.toString(shouldProxy));
+    obtainTokens = getSysProps().getBoolean(OBTAIN_BINARY_TOKEN, false);
+
+    if (shouldProxy) {
+      getLog().info("Initiating hadoop security manager.");
+      try {
+        hadoopSecurityManager = HadoopJobUtils.loadHadoopSecurityManager(getSysProps(), log);
+      } catch (RuntimeException e) {
+        throw new RuntimeException("Failed to get hadoop security manager!" + e);
+      }
+    }
+  }
+
+  @Override
+  public void run() throws Exception {
+    HadoopConfigurationInjector.prepareResourcesToInject(getJobProps(), getWorkingDirectory());
+
+    File tokenFile = null;
+    if (shouldProxy && obtainTokens) {
+      userToProxy = getJobProps().getString(USER_TO_PROXY);
+      getLog().info("Need to proxy. Getting tokens.");
+      // get tokens in to a file, and put the location in props
+      Props props = new Props();
+      props.putAll(getJobProps());
+      props.putAll(getSysProps());
+      tokenFile = HadoopJobUtils.getHadoopTokens(hadoopSecurityManager, props, getLog());
+      getJobProps().put("env." + HADOOP_TOKEN_FILE_LOCATION, tokenFile.getAbsolutePath());
+    }
+
+    try {
+      super.run();
+    } catch (Throwable t) {
+      t.printStackTrace();
+      getLog().error("caught error running the job");
+      throw new Exception(t);
+    } finally {
+      if (tokenFile != null) {
+        HadoopJobUtils.cancelHadoopTokens(hadoopSecurityManager, userToProxy, tokenFile, getLog());
+        if (tokenFile.exists()) {
+          tokenFile.delete();
+        }
+      }
+    }
+  }
+
+  @Override
+  protected String getJavaClass() {
+    return HADOOP_SECURE_SPARK_WRAPPER;
+  }
+
+  @Override
+  protected String getJVMArguments() {
+    String args = super.getJVMArguments();
+
+    String typeUserGlobalJVMArgs = getJobProps().getString("jobtype.global.jvm.args", null);
+    if (typeUserGlobalJVMArgs != null) {
+      args += " " + typeUserGlobalJVMArgs;
+    }
+    String typeSysGlobalJVMArgs = getSysProps().getString("jobtype.global.jvm.args", null);
```

Comment:
I've done the next best thing, which is to put it in the HadoopJobUtils.

This way at least the HadoopPigJob, HadoopHiveJob will also be able to reuse it